accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+because the certificate is only valid for three days a user will probably always need to update their certificate on their 
+account page. I would like to propose a set of changes to improve the experience for users that sign their NuGet package
+with Trusted Signing.
+
+## Explanation
+
+### Functional explanation
+
+### Step 1: Register EKU of certificate that was added by Trusted Signing
+
+#### How would this be implemented?
+
+The first step would be to store the EKU that is inside the certificate that was used to sign the NuGet package with Trusted
+Signing. The certificate of Tursted Signing is only valid for thee days but it contains an EKU to that can be used to get the
+identity of the owner of the certificate. There are two EKU's in the certificate. One that tells us that it was signed with
+Trusted Signing (`1.3.6.1.4.1.311.97.1.0`) and another one that contains the Public Trust identity (e.g. 
+`1.3.6.1.4.1.311.97.990309390.766961637.194916062.941502583`). More details about this can be found here:
+https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-cert-management. 
+
+I would like to propose to link EKU's to the user in a similar way as that is now done with certificates:
+
+![EKU](images/trusted-signing-eku.png)
+
+But his would require a user to enter that EKU in the interface to link it with their account. We could also start with
+automatically linking the EKU to the user by using the current "certificate upload" interface. The user would need to do the
+following:
+- Extract the `.cer` file of the signing certificate from the NuGet package.
+- Upload the `.cer` file and link it to their account.
+- Upload the NuGet package that was signed with Trusted Signing.
+
+Below is a picture that show how the certificate and EKU would be linked to user before and after the NuGet package is uploaded.
+
+![EKU link](images/trusted-signing-eku-link.png)
+
+This would mean that the certificate is no longer linked to the user and will no longer be shown in the interface because the EKU
+is now linked to the user. That will mean that the user would need to contact the NuGet team to remove the EKU link until updates
+to the interface have been made. But I don't think a lot of people will need this functionality right away.
+
+The EKU will only be linked when the following conditions are true:
+- The package was signed within the last X days where X should be decided by the NuGet team (not sure if this is really necessary).
+- The signing certificate is signed by `Microsoft Identity Verification Root Certificate Authority 2020`.
+  (https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-trust-models)
+- The signing certificate has a valid counter-signature (timestamp).
+- The signing certificate contains the Trusted Signing EKU.
+- The signing certificate contains a Public Trust Identity EKU.
+- The Public Trust Identity EKU is not already linked to the user.
```

Comment:
To be clear, we will allow an EKU to be linked to multiple users/orgs right? This is possible today since the same .cer file can be uploaded by multiple users and this means the same fingerprint is possible. 

I think it is okay to allow this because a) both users/orgs would need access to signing permission for this to matter and b) there may be legitimate scenarios for this, such as one signing identity (TS account) but multiple NuGet.org profiles (one user with multiple orgs).

Reply:
The unique EKU for the validated identity in Trusted Signing is tied to the Subscription, so other accounts and certificate profiles will be able to use that unique EKU. This allows orgs to organize their account resources with their business/development units as they see fit. 

Please consider that the signing certs will not be signed by the root directly, Microsoft Identity Verification Root Certificate Authority 2020, but its issuer will be chained to the Microsoft ID Verified Code Signing PCA 2021 that is signed by the root.

Reply:
I have updated the PR to make it more clear that it should be the root CA.